### DISCUSSION TITLE
Write string data to cube

### DISCUSSION THREAD

**Discussion by FelixStuyck** [cubewise-code/tm1py#757] [Q&A] [answered]
Hello all,

Is it possible to write string data to a cube?
Currently I have not succeeded using the write functions cells.write_dataframe or cubes.cells.write.
I haven't found it in the documentation.

Thanks,
Felix

**Comment by macsir** [comment]
Could you post some errors when trying to use cells.write_dataframe or cubes.cells.write? I believe they support string data. Thanks.

**Comment by FelixStuyck** [comment]
Error message:
TM1pyRestException: Text: '{"error":{"code":"143","message":"DataSpreadFailed"}}' - Status Code: 400 - Reason: 'Bad Request' - Headers: {'Content-Length': '70', 'Connection': 'keep-alive', 'Content-Encoding': 'gzip', 'Cache-Control': 'no-cache', 'Content-Type': 'application/json; charset=utf-8', 'OData-Version': '4.0'}

**Comment by macsir** [comment]
try this?[https://www.ibm.com/support/pages/data-spread-failed](url)
![image](https://user-images.githubusercontent.com/3345149/177308186-773ff048-82d6-4bd7-8eea-13257d4b830a.png)

**Reply by FelixStuyck** [reply]
I've set this to grant, but the message remains the same.

**Comment by MariusWirtz** [comment, marked as answer]
@FelixStuyck
I think you get this error message because you are attempting to write string data to numeric measures.

**Reply by macsir** [reply]
Agree with @MariusWirtz , could be the reason.
Please start with the simplest cube. e.g. two dimensions with several elements in each and test from there.

**Reply by FelixStuyck** [reply]
You're right the element was defined as numeric. I can't believe I've missed that.
Thank you
